cpp/src/io/utilities/config_utils.cpp

```diff
@@ -80,4 +88,38 @@ bool is_stable_enabled() { return is_all_enabled() or get_env_policy() == usage_
 
 }  // namespace nvcomp_integration
 
-}  // namespace cudf::io::detail
+inline std::mutex& host_mr_lock()
+{
+  static std::mutex map_lock;
+  return map_lock;
+}
+
+inline rmm::host_async_resource_ref default_pinned_mr()
+{
+  static rmm::mr::pinned_host_memory_resource default_mr{};
+  return default_mr;
+}
+
+CUDF_EXPORT inline auto& host_mr()
+{
+  static rmm::host_async_resource_ref host_mr = default_pinned_mr();
+  return host_mr;
+}
+
+}  // namespace detail
+
+rmm::host_async_resource_ref set_host_memory_resource(rmm::host_async_resource_ref mr)
```

Comment:
Now I realize why the header mirrors this file's name :)
We should still probably pick a better name for the public header (and I can say "better", I named this file! :D ).

Reply:
Any particular opinions?  It does cover a number of things beyond just the host allocator.